Libraries/LibC/netdb.cpp

```diff
@@ -257,22 +257,64 @@ struct servent* getservent()
     }
     
     if(!keep_service_file_open) {
-        fclose(services_file);
-        services_file = nullptr;
+        endservent();
     }
-
     return service_entry;
 
 }
 struct servent* getservbyname(const char* name, const char* protocol)
 {
-    dbg() << "FIXME: getservbyname(\"" << name << "\", \"" << protocol << "\")";
-    ASSERT_NOT_REACHED();
+    int previous_file_open_setting = keep_service_file_open;
+    setservent(1);
+    struct servent* current_service = nullptr;
+    if(!protocol) {
+        do {
+            current_service = getservent();
+            if(!current_service) {
+                return nullptr;
+            }
+        } while(strcmp(current_service->s_name, name) != 0);
+    }
+    else {
+        do {
+            current_service = getservent();
+            if(!current_service) {
+                return nullptr;
```

Comment:
Should we close the services file here, if it wasn't in "stay open" mode?
Can we use a `ScopeGuard` for this instead?